kedro-datasets/kedro_datasets/pandas/sql_dataset.py

```diff
@@ -212,29 +214,33 @@ def __init__(  # noqa: PLR0913
         self._save_args["name"] = table_name
 
         self._connection_str = credentials["con"]
-        self.create_connection(self._connection_str)
 
         self.metadata = metadata
 
     @classmethod
-    def create_connection(cls, connection_str: str) -> None:
+    def create_connection(cls, connection_str: str):
```

Comment:
Just a quick question out of curiosity. I noticed `create_connection` has a `cls` argument, but it is used in the `engine` method (property), which is called with the `self` argument. I know that Python isn't too picky about arguments (if I remember correctly you can name `self` of `cls` whatever you want), but I would expect to have matching arguments. So either `create_connection` to have the `self` argument or `engine` to have `cls` argument...